typescript/test/CodeGeneratorTest.ts

```diff
@@ -0,0 +1,50 @@
+const execa = require('execa');
+const util = require('util');
+const path = require('path');
+const fs = require('fs');
+
+const SHOWCASE_CLIENT_LIB = './showcase';
+const TMP_CLIENT_LIB = './showcase/tmp';
```

Comment:
Let's use `path.join` everywhere to make it compatible across different OS (Windows is a real use case). `path.join(cwd, 'showcase', 'tmp');` where `cwd = process.cwd()` - here and below. There should be no forward slashes anywhere in the code (Windows uses backslashes).